### ISSUE TITLE
`allow_implicit_indents` doesn't support `QUALIFY`

### ISSUE THREAD

**Issue by mroy-seedbox** [sqlfluff/sqlfluff#6325]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

For the following query, [LT02](https://docs.sqlfluff.com/en/stable/reference/rules.html#rule-LT02) is triggered as expected: `Expected line break and indent of 2 spaces before '1'. sqlfluff(LT02: layout.indent)`

```sql
SELECT 123
WHERE 1 = 1
  AND 2 = 2
QUALIFY 1 = 1
  AND 2 = 2;
```

But when setting `allow_implicit_indents = True`, the issue disappears for the `WHERE`, but remains for the `QUALIFY`:

![image](https://github.com/user-attachments/assets/3882a9a6-c8be-4b50-b9c5-d4480d6cb331)

### Expected Behaviour

No issue is detected.

### Observed Behaviour

An issue is detected.

### How to reproduce

See above (plus the minimal configurations below).

### Dialect

Snowflake (but should apply to any dialect).

### Version

```
==== sqlfluff ====
sqlfluff:                3.2.2 python:                 3.10.7
implementation:        cpython verbosity:                   1
dialect:             snowflake templater:               jinja
rules:                                all
```

### Configuration

```
[sqlfluff]
templater = jinja
dialect = snowflake

[sqlfluff:indentation]
tab_space_size = 2
allow_implicit_indents = True
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

I'm not sure where to start. I think this one might be beyond me.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by mroy-seedbox** [comment]
The same issue also happens with functions, like `IFF()`:

```sql
IFF(condition,
  column1,
  column2
)
```

Which must currently be updated to:

```sql
IFF(
  condition,
  column1,
  column2
)
```

**Comment by mroy-seedbox** [comment]
The same issue also happens when using `FROM TABLE(...) in Snowflake:
![image](https://github.com/user-attachments/assets/bb67edcf-6986-4665-97de-6ada6bbeb591)

Or:
![image](https://github.com/user-attachments/assets/7d7edc09-3445-4b65-94d7-ca92a0e38100)
